File: src/angular/src/app/proxy/settings/dtos/job-title-dto.ts
import { AuditedEntityDto } from '@abp/ng.core';

export interface JobTitleDto extends AuditedEntityDto<string> {
    nameAr: string;
    nameEn?: string;
    description?: string;
    departmentId?: string;
    departmentName?: string;
}

export interface CreateUpdateJobTitleDto {
    nameAr: string;
    nameEn?: string;
    description?: string;
    departmentId?: string;
}

File: src/src/HIS.Application.Contracts/Settings/Dtos/JobTitleDto.cs
using System;
using System.ComponentModel.DataAnnotations;
using Volo.Abp.Application.Dtos;

namespace HIS.Settings.Dtos;

public class JobTitleDto : AuditedEntityDto<Guid>
{
    public string NameAr { get; set; }
    public string NameEn { get; set; }
    public string Description { get; set; }
    public Guid? DepartmentId { get; set; }
    public string DepartmentName { get; set; } // For display
}

public class CreateUpdateJobTitleDto
{
    [Required]
    [StringLength(100)]
    public string NameAr { get; set; }

    [StringLength(100)]
    public string NameEn { get; set; }

    [StringLength(500)]
    public string Description { get; set; }

    public Guid? DepartmentId { get; set; }
}
